### PR TITLE
SER-1546 purge thumbnails endpoint

### PR DESCRIPTION
**Pull request by idradm**
https://wikia-inc.atlassian.net/browse/SER-1546

### CODE REVIEW

**Comment by idradm** [comment]
Be aware, work in progress.

File: src/vignette/http/route_helpers.clj [hunk 1]
```diff
@@ -28,6 +31,25 @@
     (create-head-response image-params)
     (error-response 404 image-params)))
 
+(defn error-catcher [request handler]
+  (try+
+    (apply handler [])
+    (catch Exception e
+      (println (.getMessage e) ":" (:uri request))
+      (perf/publish {:exception-count 1})
+      (log/warn (str e) {:path  (:uri request)
```
Comment (tmszdmsk) [inline src/vignette/http/route_helpers.clj]: do we need both `println` and `log`?

Comment (idradm) [reply src/vignette/http/route_helpers.clj]: println goes to consul output only, where log goes to kibana and is not outputted to consol on default

File: test/vignette/http/route_helpers_test.clj [hunk 2]
```diff
@@ -65,6 +65,22 @@
       (sp/get-thumbnail ..store.. original-forced-image-params) => nil
       (sp/get-original ..store.. original-forced-image-params) => nil)))
 
+(facts :handle-delete
```
Comment (tmszdmsk) [inline test/vignette/http/route_helpers_test.clj]: 👍

File: src/vignette/http/route_helpers.clj [hunk 3]
```diff
@@ -28,6 +31,25 @@
     (create-head-response image-params)
     (error-response 404 image-params)))
 
+(defn error-catcher [request handler]
```
Comment (slayful) [inline src/vignette/http/route_helpers.clj]: Add comment why non-default error catcher is used.

File: src/vignette/storage/s3.clj [hunk 4]
```diff
@@ -113,13 +113,17 @@
...
+    (perf/timing :s3-delete (s3/delete-object creds bucket path)))
+
   (object-exists? [this bucket path]
-    (s3/object-exists? (add-timeouts :head (:creds this))
-                       bucket
-                       path))
+    (try
+      (not (empty? (s3/list-objects creds bucket {:prefix path :max-keys 1})))
+      (catch AmazonS3Exception e
+        (if (= (.getErrorCode e) "AccessDenied") false (throw e)))))
```
Comment (slayful) [inline src/vignette/storage/s3.clj]: please add a comment why.

File: test/vignette/http/route_helpers_test.clj [hunk 5]
```diff
@@ -65,6 +65,22 @@
...
+  (let [original-image-params {}]
+    (handle-delete ..store.. original-image-params ..request..) =>
+    {:body "Bad Request", :headers {"X-Thumbnailer" "Vignette"}, :status 400}
```
Comment (slayful) [inline test/vignette/http/route_helpers_test.clj]: This prolly should be forbidden or unauthorized, whatever we do in Pandroa.

File: test/vignette/storage/s3_test.clj [hunk 6]
```diff
@@ -64,10 +64,18 @@
 (facts :s3 :object-exists
   (object-exists? (create-s3-storage-system ..creds..) ..bucket.. ..path..) => true
   (provided
-    (add-timeouts :head ..creds..) => ..timeout-creds..
-    (s3/object-exists? ..timeout-creds.. ..bucket.. ..path..) => true)
+    (s3/list-objects ..creds.. ..bucket.. {:prefix ..path.. :max-keys 1}) => {:test 1})
```
Comment (slayful) [inline test/vignette/storage/s3_test.clj]: `{:non-empty-map 1}`

File: test/vignette/storage/static_assets_test.clj [hunk 7]
```diff
@@ -103,3 +103,35 @@
...
+      (list-objects store "abcd" "images/thumb/d6f0194a-ea6a-410d-9c45-81411b43abcd/") => {})
+
+    (delete-thumbnails (sa/create-static-image-storage store --static-asset-get--)
+      {:options {}, :image-type "images", :uuid "d6f0194a-ea6a-410d-9c45-81411b43abcd"}) => true
+    (provided
+      (object-exists? store "abcd" "images/thumb/d6f0194a-ea6a-410d-9c45-81411b43abcd/") => true
+      (list-objects store "abcd" "images/thumb/d6f0194a-ea6a-410d-9c45-81411b43abcd/") => {:objects [{:key "images/thumb/d6f0194a-ea6a-410d-9c45-81411b43abcd/test-thumb"}]}
+      (delete-object store "abcd" "images/thumb/d6f0194a-ea6a-410d-9c45-81411b43abcd/test-thumb") => nil)
+
+
```
Comment (slayful) [inline test/vignette/storage/static_assets_test.clj]: too many newlines

**Pull request merged**